src/nvim/linematch.c

```diff
@@ -143,15 +147,17 @@ static int count_n_matched_chars(const char **sp, const size_t n, bool iwhite)
   return matched_chars;
 }
 
-void fastforward_buf_to_lnum(const char **s, linenr_T lnum)
+mmfile_t fastforward_buf_to_lnum(mmfile_t s, linenr_T lnum)
 {
   for (int i = 0; i < lnum - 1; i++) {
-    *s = strchr(*s, '\n');
-    if (!*s) {
-      return;
+    s.ptr = strnchr(s.ptr, (size_t *)&s.size, '\n');
```

Comment:
coverity warning looks valid, worth an assert?

```
CID 510275:  Memory - illegal accesses  (INCOMPATIBLE_CAST)
/src/nvim/linematch.c: 153 in fastforward_buf_to_lnum()
147       return matched_chars;
148     }
149     
150     mmfile_t fastforward_buf_to_lnum(mmfile_t s, linenr_T lnum)
151     {
152       for (int i = 0; i < lnum - 1; i++) {
>>>     CID 510275:  Memory - illegal accesses  (INCOMPATIBLE_CAST)
>>>     Pointer "&s.size" points to an object whose effective type is "int" (32 bits, signed) but is dereferenced as a wider "unsigned long" (64 bits, unsigned). This may lead to memory corruption.
153         s.ptr = strnchr(s.ptr, (size_t *)&s.size, '\n');
154         if (!s.ptr) {
155           break;
156         }
157         s.ptr++;
158         s.size--;
```